persist/fs/commitlog/iterator.go

```diff
@@ -95,6 +95,7 @@ func (i *iterator) Next() bool {
 	var err error
 	i.read.series, i.read.datapoint, i.read.unit, i.read.annotation, err = i.reader.Read()
 	if err == io.EOF {
+		i.reader.Close()
```

Comment:
Good catch, is this only place where we close readers?  Maybe we should audit this code a little for these cases.